### PR TITLE
Scan: Remove chainID hardcoded

### PR DESCRIPTION
**Pull request by evilpeach**
Review this PR after #1472 has merged.

- Remove hardcoded in `createRawTx`

### CODE REVIEW

**Review by tansawit — CHANGES_REQUESTED** [review]
Right now there is a loading state every time the green connect button is pressed (because the chainID is being retrieved every time). Ideally the chainID should only be fetched once during the first connect button press.

**Review by sorawit — CHANGES_REQUESTED** [review]
Agree with @tansawit . Let's make the green connect passes around `ChainID` and not load it in the modal components.

File: scan/src/components/UserAccount.re [hunk 1]
```diff
@@ -179,30 +179,35 @@ module Balance = {
...
-        <Col> <Balance address /> </Col>
-        <Col> <HSpacing size={`px(5)} /> </Col>
-        <Col> <FaucetBtn address={address->Address.toBech32} /> </Col>
-      </Row>
-      <VSpacing size=Spacing.md />
-      <Row justify=Row.Right> <Col> <SubmitTxBtn submitTx /> </Col> </Row>
-    </>
-  | None => <Col> <ConnectBtn connect /> </Col>
+  switch (metadataSub) {
+  | Data({chainID}) =>
```
Comment (sorawit) [inline scan/src/components/UserAccount.re]: Why don't we switch on `accountOpt` first? So we always show disconnect button even when we don't have chainID?

File: scan/src/components/UserAccount.re [hunk 2]
```diff
@@ -179,30 +179,35 @@ module Balance = {
...
+          <Col> <Balance address /> </Col>
+          <Col> <HSpacing size={`px(5)} /> </Col>
+          <Col> <FaucetBtn address={address->Address.toBech32} /> </Col>
+        </Row>
+        <VSpacing size=Spacing.md />
+        <Row justify=Row.Right> <Col> <SubmitTxBtn submitTx /> </Col> </Row>
+      </>
+    | None => <Col> <ConnectBtn connect={_ => connect(chainID)} /> </Col>
+    }
+  | _ => React.null
```
Comment (sorawit) [inline scan/src/components/UserAccount.re]: Can we do the loading animation in this PR? How hard is it?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/components/UserAccount.re (hunk 1 above)
Comment (evilpeach) [reply scan/src/components/UserAccount.re]: If we don't have chainID, we show nothing.

File: scan/src/components/UserAccount.re (hunk 2 above)
Comment (evilpeach) [reply scan/src/components/UserAccount.re]: No, I'll add it.

File: scan/src/components/UserAccount.re (hunk 1 above)
Comment (evilpeach) [reply scan/src/components/UserAccount.re]: Oh, my bad. just push edited version.

**Review by sorawit — APPROVED** [review]

**Review by tansawit — APPROVED** [review]

**Pull request merged**